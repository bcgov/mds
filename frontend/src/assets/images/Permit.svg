<svg width="170" height="188" viewBox="0 0 170 188" fill="none" xmlns="http://www.w3.org/2000/svg">
<path d="M90.5048 0H33.8445C26.0537 0 19.7503 6.38418 19.7503 14.1871L19.6794 127.684C19.6794 135.486 25.9829 141.871 33.7737 141.871H118.835C126.626 141.871 133 135.486 133 127.684V42.5612L90.5048 0ZM104.67 113.496H48.0096V99.3094H104.67V113.496ZM104.67 85.1224H48.0096V70.9353H104.67V85.1224ZM83.4223 49.6547V10.6403L122.376 49.6547H83.4223Z" fill="#B9ADA2"/>
<path d="M21.4636 6.49789H80.5028L124.782 50.7485V139.25C124.782 147.362 118.14 154 110.022 154H21.3898C13.272 154 6.70386 147.362 6.70386 139.25L6.77766 21.2481C6.77766 13.1355 13.3458 6.49789 21.4636 6.49789Z" fill="#E9E9E9"/>
<path d="M70.8253 11.5879H14.1651C6.37428 11.5879 0.070827 17.972 0.070827 25.7749L0 139.271C0 147.074 6.30345 153.458 14.0942 153.458H99.1555C106.946 153.458 113.321 147.074 113.321 139.271V54.149L70.8253 11.5879ZM84.9904 125.084H28.3301V110.897H84.9904V125.084ZM84.9904 96.7102H28.3301V82.5232H84.9904V96.7102ZM63.7428 61.2426V22.2282L102.697 61.2426H63.7428Z" fill="#B9ADA2"/>
<g filter="url(#filter0_d)">
<circle cx="116" cy="134" r="38" fill="#BC2929"/>
</g>
<line x1="97.1159" y1="116.149" x2="136.636" y2="150.349" stroke="white" stroke-width="5"/>
<line x1="98.2721" y1="151.992" x2="135.228" y2="115.037" stroke="white" stroke-width="5"/>
<defs>
<filter id="filter0_d" x="62" y="80" width="108" height="108" filterUnits="userSpaceOnUse" color-interpolation-filters="sRGB">
<feFlood flood-opacity="0" result="BackgroundImageFix"/>
<feColorMatrix in="SourceAlpha" type="matrix" values="0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 127 0"/>
<feOffset/>
<feGaussianBlur stdDeviation="8"/>
<feColorMatrix type="matrix" values="0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0.25 0"/>
<feBlend mode="normal" in2="BackgroundImageFix" result="effect1_dropShadow"/>
<feBlend mode="normal" in="SourceGraphic" in2="effect1_dropShadow" result="shape"/>
</filter>
</defs>
</svg>
