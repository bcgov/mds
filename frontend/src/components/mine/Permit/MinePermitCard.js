import React from "react";
import PropTypes from "prop-types";
import CustomPropTypes from "@/customPropTypes";
import { formatTitleString, formatDate } from "@/utils/helpers";

const propTypes = {
  permit: PropTypes.objectOf(CustomPropTypes.permit),
  PartyRelationships: PropTypes.arrayOf(CustomPropTypes.partyRelationship),
};

const defaultProps = {
  permit: {},
  PartyRelationships: [],
};

export const PermitCard = (props) => {
  const pmt = props.PartyRelationships.filter((pr) => pr.mine_party_appt_type_code === "PMT")
    .filter((pmts) => pmts.related_guid.includes(props.permit.permit_guid))
    .sort((a, b) => new Date(b.start_date) - new Date(a.start_date))[0];

  return (
    <div>
      <h4>{formatTitleString(props.permit.permit_no)}</h4>
      <br />
      <h6>Last Amended</h6>
      <span>
        {props.permit.amendments[0] ? (
          formatDate(props.permit.amendments[0].issue_date)
        ) : (
          <i>No Amendments</i>
        )}
      </span>
      <br />
      <br />
      <h6>Permittee</h6>
      <span>{pmt ? pmt.party.name : <i>None Assigned</i>}</span>
      <br />
      <br />
    </div>
  );
};

PermitCard.propTypes = propTypes;
PermitCard.defaultProps = defaultProps;

export default PermitCard;
