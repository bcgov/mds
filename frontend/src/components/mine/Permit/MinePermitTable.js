import React from "react";
import { Table, Menu, Dropdown, Button, Icon } from "antd";
import NullScreen from "@/components/common/NullScreen";
import AuthorizationWrapper from "@/components/common/wrappers/AuthorizationWrapper";
import * as Strings from "@/constants/strings";
import * as Permission from "@/constants/permissions";
import CustomPropTypes from "@/customPropTypes";
import { formatDate } from "@/utils/helpers";
import PropTypes from "prop-types";
import { connect } from "react-redux";
import { getPartyRelationships } from "@/selectors/partiesSelectors";
import { getDropdownPermitStatusOptions } from "@/selectors/staticContentSelectors";
import { BRAND_PENCIL, EDIT, EDIT_OUTLINE, CARAT } from "@/constants/assets";
import downloadFileFromDocumentManager from "@/utils/actionlessNetworkCalls";

/**
 * @class  MinePermitTable - displays a table of permits and permit amendments
 */

const amalgamtedPermit = "ALG";

const propTypes = {
  permits: PropTypes.arrayOf(CustomPropTypes.permit).isRequired,
  partyRelationships: PropTypes.arrayOf(CustomPropTypes.partyRelationship),
  permitStatusOptions: PropTypes.arrayOf(CustomPropTypes.dropdownListItem).isRequired,
  major_mine_ind: PropTypes.bool.isRequired,
  openEditPermitModal: PropTypes.func.isRequired,
  openEditAmendmentModal: PropTypes.func.isRequired,
  openAddPermitAmendmentModal: PropTypes.func.isRequired,
  openAddAmalgamatedPermitModal: PropTypes.func.isRequired,
};

const defaultProps = {
  partyRelationships: [],
};

const renderDocumentLink = (file, text) => (
  <a
    key={file.mine_document_guid}
    onClick={() => downloadFileFromDocumentManager(file.document_manager_guid, file.document_name)}
  >
    {text}
  </a>
);

const renderPermitNo = (permit) => {
  const permitNoShouldLinkToDocument =
    permit.amendments[0] &&
    permit.amendments[0].permit_amendment_type_code === amalgamtedPermit &&
    permit.amendments[0].related_documents[0];
  return permitNoShouldLinkToDocument
    ? renderDocumentLink(permit.amendments[0].related_documents[0], permit.permit_no)
    : permit.permit_no;
};

const columns = [
  {
    title: "Permit No.",
    dataIndex: "permitNo",
    key: "permitNo",
    render: (text, record) => <div title="Permit No.">{renderPermitNo(record.permit)}</div>,
  },
  {
    title: "Status",
    dataIndex: "status",
    key: "status",
    render: (text, record) => (
      <div title="Status">
        {record.permitStatusOptions.find((item) => item.value === text).label}
      </div>
    ),
  },
  {
    title: "Permittee",
    dataIndex: "permittee",
    key: "permittee",
    render: (text) => <div title="Permittee">{text}</div>,
  },
  {
    title: "Authorization End Date",
    dataIndex: "authorizationEndDate",
    key: "authorizationEndDate",
    render: (text) => <div title="Authorization End Date">{text}</div>,
  },

  {
    title: "First Issued",
    dataIndex: "firstIssued",
    key: "firstIssued",
    render: (text) => <div title="First Issued">{text}</div>,
  },
  {
    title: "Last Amended",
    dataIndex: "lastAmended",
    key: "lastAmended",
    render: (text) => <div title="Last Amended">{text}</div>,
  },
  {
    title: "",
    dataIndex: "addEditButton",
    key: "addEditButton",
    align: "right",
    render: (text, record) => {
      const menu = (
        <Menu>
          <Menu.Item key="0">
            <button
              type="button"
              className="full add-permit-dropdown-button"
              onClick={(event) => record.openAddAmalgamatedPermitModal(event, record.permit)}
            >
              <div>
                <Icon
                  type="plus"
                  className="padding-small add-permit-dropdown-button-icon"
                  theme="outlined"
                />
                {text.hasAmalgamated ? "Add permit amendment" : "Amalgamate permit"}
              </div>
            </button>
          </Menu.Item>
          {!text.hasAmalgamated && (
            <Menu.Item key="1">
              <button
                type="button"
                className="full add-permit-dropdown-button"
                onClick={(event) => record.openAddPermitAmendmentModal(event, record.permit)}
              >
                <div>
                  <Icon
                    type="plus"
                    className="padding-small add-permit-dropdown-button-icon"
                    theme="outlined"
                  />
                  Add permit amendment
                </div>
              </button>
            </Menu.Item>
          )}
          <Menu.Item key="2">
            <button
              type="button"
              className="full"
              onClick={(event) =>
                record.openEditPermitModal(event, record.permit, record.description)
              }
            >
              <img
                alt="document"
                className="padding-small"
                src={BRAND_PENCIL}
                style={{ paddingRight: "15px" }}
              />
              Edit permit status
            </button>
          </Menu.Item>
        </Menu>
      );
      return (
        <AuthorizationWrapper permission={Permission.CREATE} isMajorMine={text.major_mine_ind}>
          <Dropdown className="full-height full-mobile" overlay={menu} placement="bottomLeft">
            <Button type="secondary" className="permit-table-button">
              <div className="padding-small">
                <img className="padding-small--right icon-svg-filter" src={EDIT} alt="Add/Edit" />
                Add/Edit
                <img
                  className="padding-small--right icon-svg-filter"
                  src={CARAT}
                  alt="Menu"
                  style={{ paddingLeft: "5px" }}
                />
              </div>
            </Button>
          </Dropdown>
        </AuthorizationWrapper>
      );
    },
  },
];

const childColumns = [
  {
    title: "Amendment",
    dataIndex: "amendmentNumber",
    key: "amendmentNumber",
    width: "130px",
    render: (text) => <div title="Amendment">{text}</div>,
  },
  {
    title: "Date Issued",
    dataIndex: "issueDate",
    key: "issueDate",
    width: "90px",
    render: (text) => <div title="Issue Date">{text}</div>,
  },
  {
    title: "Description",
    dataIndex: "description",
    key: "description",
    render: (text) => (
      <div title="Description">
        <p className="wrapped-text" style={{ maxWidth: "800px" }}>
          {text}
        </p>
      </div>
    ),
  },
  {
    title: "Files",
    dataIndex: "documents",
    key: "documents",
    render: (text) => (
      <div title="Files">
        <ul>
          {text.map((file) => (
            <li className="wrapped-text">{renderDocumentLink(file, file.document_name)}</li>
          ))}
        </ul>
      </div>
    ),
  },
  {
    title: "",
    dataIndex: "amendmentEdit",
    key: "amendmentEdit",
    align: "right",
    render: (text, record) => (
      <AuthorizationWrapper permission={Permission.CREATE} isMajorMine={text.major_mine_ind}>
        <Button
          className="permit-table-button"
          type="ghost"
          onClick={(event) => record.openEditAmendmentModal(event, text.amendment, record.permit)}
        >
          <div>
            <img className="padding-small--right icon-svg-filter" src={EDIT_OUTLINE} alt="Edit" />
          </div>
        </Button>
      </AuthorizationWrapper>
    ),
  },
];

const getPermittees = (partyRelationships, permit) =>
  partyRelationships
    .filter(({ related_guid }) => permit.permit_guid === related_guid)
    .sort((order1, order2) => {
      const date1 = Date.parse(order1.due_date) || 0;
      const date2 = Date.parse(order2.due_date) || 0;
      return date1 === date2 ? order1.order_no - order2.order_no : date1 - date2;
    });

const getPermitteeName = (permittees) =>
  permittees[0] ? permittees[0].party.party_name : Strings.EMPTY_FIELD;

const transformRowData = (
  permit,
  partyRelationships,
  major_mine_ind,
  openEditPermitModal,
  openAddPermitAmendmentModal,
  openAddAmalgamatedPermitModal,
  permitStatusOptions
) => {
  const latestAmendment = permit.amendments[0];
  const firstAmendment = permit.amendments[permit.amendments.length - 1];

  const permittees = getPermittees(partyRelationships, permit);
  const permitteeName = partyRelationships.length === 0 ? "" : getPermitteeName(permittees);
  const hasAmalgamated = permit.amendments.find(
    (pa) => pa.permit_amendment_type_code === amalgamtedPermit
  );

  return {
    key: permit.permit_guid,
    lastAmended: (latestAmendment && formatDate(latestAmendment.issue_date)) || Strings.EMPTY_FIELD,
    permitNo: permit.permit_no || Strings.EMPTY_FIELD,
    firstIssued: (firstAmendment && formatDate(firstAmendment.issue_date)) || Strings.EMPTY_FIELD,
    permittee: permitteeName,
    authorizationEndDate:
      (latestAmendment && formatDate(latestAmendment.authorization_end_date)) ||
      Strings.EMPTY_FIELD,
    amendments: permit.amendments,
    status: permit.permit_status_code,
    addEditButton: {
      major_mine_ind,
      hasAmalgamated,
    },
    openEditPermitModal,
    openAddPermitAmendmentModal,
    openAddAmalgamatedPermitModal,
    permitStatusOptions,
    permit,
  };
};

const transformChildRowData = (
  amendment,
  record,
  amendmentNumber,
  major_mine_ind,
  openEditAmendmentModal
) => ({
  amendmentNumber,
  amendmentType: amendment.permit_amendment_type_code,
  receivedDate: formatDate(amendment.received_date) || Strings.EMPTY_FIELD,
  issueDate: formatDate(amendment.issue_date) || Strings.EMPTY_FIELD,
  authorizationEndDate: formatDate(amendment.authorization_end_date) || Strings.EMPTY_FIELD,
  description: amendment.description || Strings.EMPTY_FIELD,
  amendmentEdit: {
    major_mine_ind,
    amendment,
  },
  openEditAmendmentModal,
  permit: record.permit,
  documents: amendment.related_documents,
});

export const MinePermitTable = (props) => {
  const amendmentHistory = (permit) => {
    const childRowData = permit.amendments.map((amendment, index) =>
      transformChildRowData(
        amendment,
        permit,
        permit.amendments.length - index,
        props.major_mine_ind,
        props.openEditAmendmentModal
      )
    );
    return (
      <Table align="left" pagination={false} columns={childColumns} dataSource={childRowData} />
    );
  };

  const rowData = props.permits.map((permit) =>
    transformRowData(
      permit,
      props.partyRelationships,
      props.major_mine_ind,
      props.openEditPermitModal,
      props.openAddPermitAmendmentModal,
      props.openAddAmalgamatedPermitModal,
      props.permitStatusOptions
    )
  );

  return (
    <Table
      className="nested-table"
      rowClassName={() => "table-row-align-middle"}
      align="left"
      pagination={false}
      columns={columns}
      dataSource={rowData}
      expandedRowRender={amendmentHistory}
      locale={{ emptyText: <NullScreen type="permit" /> }}
    />
  );
};

const mapStateToProps = (state) => ({
  partyRelationships: getPartyRelationships(state),
  permitStatusOptions: getDropdownPermitStatusOptions(state),
});

MinePermitTable.propTypes = propTypes;
MinePermitTable.defaultProps = defaultProps;

export default connect(mapStateToProps)(MinePermitTable);
