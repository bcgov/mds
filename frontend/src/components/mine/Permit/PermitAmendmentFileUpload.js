import React from "react";
import PropTypes from "prop-types";
import { Field } from "redux-form";
import { Form } from "antd";

import { PERMIT } from "@/constants/API";
import FileUpload from "@/components/common/FileUpload";
import { DOCUMENT, EXCEL } from "@/constants/fileTypes";

const propTypes = {
  onFileLoad: PropTypes.func.isRequired,
  onRemoveFile: PropTypes.func.isRequired,
  mineGuid: PropTypes.string.isRequired,
  allowMultiple: PropTypes.bool.isRequired,
};

export const PermitAmendmentFileUpload = (props) => (
  <Form.Item>
    <Field
      id="fileUpload"
      name="fileUpload"
      component={FileUpload}
      uploadUrl={`${PERMIT()}/amendments/documents?mine_guid=${props.mineGuid}`}
      acceptedFileTypesMap={{ ...DOCUMENT, ...EXCEL }}
      onFileLoad={props.onFileLoad}
      onRemoveFile={props.onRemoveFile}
      allowRevert
      allowMultiple={props.allowMultiple}
    />
  </Form.Item>
);

PermitAmendmentFileUpload.propTypes = propTypes;

export default PermitAmendmentFileUpload;
