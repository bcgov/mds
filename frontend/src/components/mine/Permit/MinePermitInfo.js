import React, { Component } from "react";
import { bindActionCreators } from "redux";
import { connect } from "react-redux";
import PropTypes from "prop-types";
import CustomPropTypes from "@/customPropTypes";
import * as Permission from "@/constants/permissions";
import AuthorizationWrapper from "@/components/common/wrappers/AuthorizationWrapper";
import {
  fetchPermits,
  createPermit,
  updatePermit,
  updatePermitAmendment,
  createPermitAmendment,
  removePermitAmendmentDocument,
} from "@/actionCreators/permitActionCreator";
import { Icon, Button } from "antd";
import MinePermitTable from "@/components/mine/Permit/MinePermitTable";
import * as ModalContent from "@/constants/modalContent";
import { modalConfig } from "@/components/modalContent/config";
import { getPermits } from "../../../reducers/permitReducer";
/**
 * @class  MinePermitInfo - contains all permit information
 */

const amalgamtedPermit = "ALG";
const originalPermit = "OGP";

const propTypes = {
  mine: CustomPropTypes.mine.isRequired,
  permits: PropTypes.arrayOf(CustomPropTypes.permit),
  partyRelationships: PropTypes.arrayOf(CustomPropTypes.partyRelationship),
  openModal: PropTypes.func.isRequired,
  closeModal: PropTypes.func.isRequired,
  createPermit: PropTypes.func.isRequired,
  fetchPermits: PropTypes.func.isRequired,
  updatePermit: PropTypes.func.isRequired,
  updatePermitAmendment: PropTypes.func.isRequired,
  createPermitAmendment: PropTypes.func.isRequired,
  removePermitAmendmentDocument: PropTypes.func.isRequired,
};

const defaultProps = {
  partyRelationships: [],
  permits: [],
};

export class MinePermitInfo extends Component {
  componentWillMount() {
    this.props.fetchPermits({ mine_guid: this.props.mine.guid });
  }

  closePermitModal = () => {
    this.props.closeModal();
    this.props.fetchPermits({ mine_guid: this.props.mine.guid });
  };

  // Permit Modals

  openAddPermitModal = (event, onSubmit, title) => {
    event.preventDefault();

    this.props.openModal({
      props: {
        initialValues: {
          mine_guid: this.props.mine.guid,
        },
        onSubmit,
        title,
        mine_guid: this.props.mine.guid,
      },
      widthSize: "50vw",
      content: modalConfig.ADD_PERMIT,
    });
  };

  openEditPermitModal = (event, permit) => {
    event.preventDefault();

    this.props.openModal({
      props: {
        initialValues: permit,
        onSubmit: this.handleEditPermit,
        title: `Edit permit status for ${permit.permit_no}`,
      },
      content: modalConfig.EDIT_PERMIT,
    });
  };

  // Permit Handlers

  handleAddPermit = (values) => {
    const payload = { mine_guid: this.props.mine.guid, ...values };

    payload.permit_no = `${values.permit_type}${values.permit_activity_type || ""}-${
      values.permit_no
    }`;

    return this.props.createPermit(payload).then(this.closePermitModal);
  };

  handleEditPermit = (values) =>
    this.props.updatePermit(values.permit_guid, values).then(this.closePermitModal);

  // Amendment Modals

  openAddAmendmentModal = (event, onSubmit, title, permit, type) => {
    event.preventDefault();
    this.props.openModal({
      props: {
        initialValues: {
          mine_guid: permit.mine_guid,
          permit_guid: permit.permit_guid,
          permit_amendment_type_code: type,
        },
        onSubmit,
        title,
        mine_guid: permit.mine_guid,
      },
      widthSize: "50vw",
      content: modalConfig.PERMIT_AMENDMENT,
    });
  };

  openEditAmendmentModal = (event, permit_amendment, permit) => {
    event.preventDefault();
    this.props.openModal({
      props: {
        initialValues: {
          ...permit_amendment,
        },
        onSubmit: this.handleEditPermitAmendment,
        title:
          permit_amendment.permit_amendment_type_code === originalPermit
            ? `Edit initial permit for ${permit.permit_no}`
            : `Edit permit amendment for ${permit.permit_no}`,
        mine_guid: permit.mine_guid,
        handleRemovePermitAmendmentDocument: this.handleRemovePermitAmendmentDocument,
      },
      widthSize: "50vw",
      content: modalConfig.PERMIT_AMENDMENT,
    });
  };

  openAddAmalgamatedPermitModal = (event, permit) =>
    this.openAddAmendmentModal(
      event,
      this.handleAddAmalgamatedPermit,
      `Add amalgamated permit to ${permit.permit_no}`,
      permit,
      amalgamtedPermit
    );

  openAddPermitAmendmentModal = (event, permit) =>
    this.openAddAmendmentModal(
      event,
      this.handleAddPermitAmendment,
      `Add permit amendment to ${permit.permit_no}`,
      permit
    );

  // Amendment Handlers

  handleEditPermitAmendment = (values) =>
    this.props
      .updatePermitAmendment(values.permit_amendment_guid, values)
      .then(this.closePermitModal);

  handleAddPermitAmendment = (values) =>
    this.props.createPermitAmendment(values.permit_guid, values).then(this.closePermitModal);

  handleAddAmalgamatedPermit = (values) =>
    this.props
      .createPermitAmendment(values.permit_guid, {
        ...values,
        permit_amendment_type_code: amalgamtedPermit,
      })
      .then(this.closePermitModal);

  handleRemovePermitAmendmentDocument = (permitAmdendmentGuid, documentGuid) =>
    this.props.removePermitAmendmentDocument(permitAmdendmentGuid, documentGuid).then(() => {
      this.props.fetchPermits({ mine_guid: this.props.mine.guid });
    });

  render() {
    return [
      <div>
        <div className="inline-flex between">
          <div />
          <div className="inline-flex between">
            <AuthorizationWrapper
              permission={Permission.CREATE}
              isMajorMine={this.props.mine.major_mine_ind}
            >
              <Button
                type="primary"
                onClick={(event) =>
                  this.openAddPermitModal(
                    event,
                    this.handleAddPermit,
                    `${ModalContent.ADD_PERMIT} to ${this.props.mine.mine_name}`
                  )
                }
              >
                <Icon type="plus" theme="outlined" style={{ fontSize: "18px" }} />
                Add a New Permit
              </Button>
            </AuthorizationWrapper>
          </div>
        </div>
      </div>,
      <br />,
      this.props.permits && (
        <MinePermitTable
          permits={this.props.permits}
          partyRelationships={this.props.partyRelationships}
          major_mine_ind={this.props.mine.major_mine_ind}
          openEditPermitModal={this.openEditPermitModal}
          openEditAmendmentModal={this.openEditAmendmentModal}
          openAddPermitAmendmentModal={this.openAddPermitAmendmentModal}
          openAddAmalgamatedPermitModal={this.openAddAmalgamatedPermitModal}
        />
      ),
    ];
  }
}

const mapStateToProps = (state) => ({
  permits: getPermits(state),
});

const mapDispatchToProps = (dispatch) =>
  bindActionCreators(
    {
      fetchPermits,
      createPermit,
      updatePermit,
      updatePermitAmendment,
      createPermitAmendment,
      removePermitAmendmentDocument,
    },
    dispatch
  );

MinePermitInfo.propTypes = propTypes;
MinePermitInfo.defaultProps = defaultProps;

export default connect(
  mapStateToProps,
  mapDispatchToProps
)(MinePermitInfo);
