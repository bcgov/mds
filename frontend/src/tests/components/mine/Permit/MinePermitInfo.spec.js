import React from "react";
import { shallow } from "enzyme";
import { MinePermitInfo } from "@/components/mine/Permit/MinePermitInfo";
import * as MOCK from "@/tests/mocks/dataMocks";

const dispatchProps = {};
const props = {};

const setupDispatchProps = () => {
  dispatchProps.fetchPermits = jest.fn();
  dispatchProps.fetchPermitStatusOptions = jest.fn();
};

const setupProps = () => {
  props.mine = MOCK.MINES.mines[MOCK.MINES.mineIds[0]];
};

beforeEach(() => {
  setupDispatchProps();
  setupProps();
});

describe("MinePermitInfo", () => {
  it("renders properly", () => {
    const component = shallow(<MinePermitInfo {...dispatchProps} {...props} />);
    expect(component).toMatchSnapshot();
  });
});
