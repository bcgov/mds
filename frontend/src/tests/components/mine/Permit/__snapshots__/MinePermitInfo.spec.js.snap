// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`MinePermitInfo renders properly 1`] = `
ShallowWrapper {
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <MinePermitInfo
    fetchPermitStatusOptions={[MockFunction]}
    fetchPermits={
      [MockFunction] {
        "calls": Array [
          Array [
            "18133c75-49ad-4101-85f3-a43e35ae989a",
          ],
        ],
      }
    }
    mine={
      Object {
        "mine_expected_documents": Array [
          Object {
            "date_created": "2018-11-19",
            "due_date": "",
            "exp_document_guid": "806608ed-d2b4-4f83-8b22-739d5223c56f",
            "exp_document_name": "OMS Manual",
            "expected_document_status": Object {
              "description": "Not Received",
              "exp_document_status_code": "MIA",
            },
            "mine_guid": "18133c75-49ad-4101-85f3-a43e35ae989a",
            "related_documents": Array [],
            "req_document_guid": "c95886bc-e4b2-4743-b38d-42eea858e9ee",
          },
        ],
        "mine_guid": "18133c75-49ad-4101-85f3-a43e35ae989a",
        "mine_location": Object {
          "latitude": null,
          "longitude": null,
        },
        "mine_name": "mine3",
        "mine_no": "BLAH9091",
        "mine_permit": Array [],
        "mine_region": "NE",
        "mine_status": Object {
          "effective_date": "2018-10-16",
          "expiry_date": null,
          "status_labels": Array [
            "Closed",
            "Care & Maintenance",
          ],
          "statusvalue": Array [
            "CLD",
            "CM",
          ],
        },
        "mine_tailings_storage_facilities": Array [
          Object {
            "mine_guid": "18133c75-49ad-4101-85f3-a43e35ae989a",
            "mine_tailings_storage_facility_guid": "e2629897-053e-4218-9299-479375e47f78",
            "mine_tailings_storage_facility_name": "MockTSF",
          },
        ],
        "mine_type": Array [
          Object {
            "mine_tenure_type_code": "PLR",
          },
          Object {
            "mine_tenure_type_code": "MIN",
          },
        ],
        "mineral_tenure_xref": Array [
          Object {
            "tenure_number_id": "1234567",
          },
        ],
        "verified_status": Object {
          "healthy_ind": null,
          "mine_guid": null,
          "mine_name": null,
          "verifying_timestamp": null,
          "verifying_user": null,
        },
      }
    }
    partyRelationships={Array []}
    permits={Array []}
  />,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateError": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "host",
    "props": Object {
      "children": <div
        className="inline-flex between"
      >
        <div />
        <div
          className="inline-flex between"
        >
          <Connect(AuthorizationWrapper)
            isMajorMine={undefined}
            permission="role_edit_permits"
          >
            <AddButton
              onClick={[Function]}
            >
              Add a New Permit
            </AddButton>
          </Connect(AuthorizationWrapper)>
        </div>
      </div>,
    },
    "ref": null,
    "rendered": Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": Array [
          <div />,
          <div
            className="inline-flex between"
          >
            <Connect(AuthorizationWrapper)
              isMajorMine={undefined}
              permission="role_edit_permits"
            >
              <AddButton
                onClick={[Function]}
              >
                Add a New Permit
              </AddButton>
            </Connect(AuthorizationWrapper)>
          </div>,
        ],
        "className": "inline-flex between",
      },
      "ref": null,
      "rendered": Array [
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {},
          "ref": null,
          "rendered": null,
          "type": "div",
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": <Connect(AuthorizationWrapper)
              isMajorMine={undefined}
              permission="role_edit_permits"
            >
              <AddButton
                onClick={[Function]}
              >
                Add a New Permit
              </AddButton>
            </Connect(AuthorizationWrapper)>,
            "className": "inline-flex between",
          },
          "ref": null,
          "rendered": Object {
            "instance": null,
            "key": undefined,
            "nodeType": "class",
            "props": Object {
              "children": <AddButton
                onClick={[Function]}
              >
                Add a New Permit
              </AddButton>,
              "isMajorMine": undefined,
              "permission": "role_edit_permits",
            },
            "ref": null,
            "rendered": Object {
              "instance": null,
              "key": undefined,
              "nodeType": "function",
              "props": Object {
                "children": "Add a New Permit",
                "onClick": [Function],
              },
              "ref": null,
              "rendered": "Add a New Permit",
              "type": [Function],
            },
            "type": [Function],
          },
          "type": "div",
        },
      ],
      "type": "div",
    },
    "type": "div",
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": <div
          className="inline-flex between"
        >
          <div />
          <div
            className="inline-flex between"
          >
            <Connect(AuthorizationWrapper)
              isMajorMine={undefined}
              permission="role_edit_permits"
            >
              <AddButton
                onClick={[Function]}
              >
                Add a New Permit
              </AddButton>
            </Connect(AuthorizationWrapper)>
          </div>
        </div>,
      },
      "ref": null,
      "rendered": Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": Array [
            <div />,
            <div
              className="inline-flex between"
            >
              <Connect(AuthorizationWrapper)
                isMajorMine={undefined}
                permission="role_edit_permits"
              >
                <AddButton
                  onClick={[Function]}
                >
                  Add a New Permit
                </AddButton>
              </Connect(AuthorizationWrapper)>
            </div>,
          ],
          "className": "inline-flex between",
        },
        "ref": null,
        "rendered": Array [
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {},
            "ref": null,
            "rendered": null,
            "type": "div",
          },
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "children": <Connect(AuthorizationWrapper)
                isMajorMine={undefined}
                permission="role_edit_permits"
              >
                <AddButton
                  onClick={[Function]}
                >
                  Add a New Permit
                </AddButton>
              </Connect(AuthorizationWrapper)>,
              "className": "inline-flex between",
            },
            "ref": null,
            "rendered": Object {
              "instance": null,
              "key": undefined,
              "nodeType": "class",
              "props": Object {
                "children": <AddButton
                  onClick={[Function]}
                >
                  Add a New Permit
                </AddButton>,
                "isMajorMine": undefined,
                "permission": "role_edit_permits",
              },
              "ref": null,
              "rendered": Object {
                "instance": null,
                "key": undefined,
                "nodeType": "function",
                "props": Object {
                  "children": "Add a New Permit",
                  "onClick": [Function],
                },
                "ref": null,
                "rendered": "Add a New Permit",
                "type": [Function],
              },
              "type": [Function],
            },
            "type": "div",
          },
        ],
        "type": "div",
      },
      "type": "div",
    },
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {},
      "ref": null,
      "rendered": null,
      "type": "br",
    },
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "class",
      "props": Object {
        "expandedRowKeys": Array [],
        "major_mine_ind": undefined,
        "onExpand": [Function],
        "openAddAmalgamatedPermitModal": [Function],
        "openAddPermitAmendmentModal": [Function],
        "openEditAmendmentModal": [Function],
        "openEditPermitModal": [Function],
        "partyRelationships": Array [],
        "permits": Array [],
      },
      "ref": null,
      "rendered": null,
      "type": [Function],
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
        "lifecycles": Object {
          "componentDidUpdate": Object {
            "onSetState": true,
          },
          "getDerivedStateFromProps": true,
          "getSnapshotBeforeUpdate": true,
          "setState": Object {
            "skipsComponentDidUpdateOnNullish": true,
          },
        },
      },
    },
    "attachTo": undefined,
    "hydrateIn": undefined,
  },
}
`;
